host/CMakeLists.txt

```diff
@@ -174,9 +174,15 @@ if (UNIX)
   target_compile_options(oehost
      PRIVATE -Wno-attributes -Wmissing-prototypes -fPIC ${PLATFORM_FLAGS}
      PUBLIC -fstack-protector-strong)
-  target_compile_definitions(oehost
-     PRIVATE _GNU_SOURCE
-     PUBLIC _FORTIFY_SOURCE=2)
+  string(TOUPPER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_SUFFIX)
```

Comment:
This is really confusing to read. Try changing it to:

```
target_compile_definitions(oehost
    PRIVATE _GNU_SOURCE
    PUBLIC $<$<CONFIG:debug>:_FORTIFY_SOURCE=2)>)
```

The `$<CONFIG:cfg>` generator expression is case insensitive, so it should be used wherever possible instead of the hack with `TOUPPER`.

Reply:
@andschwa thanks for the suggestion.

I updated the commit, can you take please another look?